### DISCUSSION TITLE
Better effect for the new Adrenaline

### DISCUSSION THREAD

**Discussion by G-M-twostay** [FakeFishGames/Barotrauma#13409] [Unstable feedback]
In an attempt to make adrenaline useful(good to see), in unstable we saw that they now give the "adrenaline rush" effect, which is a bit too powerful and hard to balance. Therefore I propose a similar effect but way easier to balance.

Provided below are the xmls for the new adrenaline and its effects.
```
<!-- Name: Adrenaline -->
<!-- A naturally occurring hormone. Useful for crafting. -->
<!-- *************************************************************************************************************************************************************************** -->
<Adrenaline name="" identifier="adrenaline" category="Medical,Material" maxstacksize="32" maxstacksizecharacterinventory="8" cargocontaineridentifier="mediccrate" Tags="smallitem,chem,medical,syringe" useinhealthinterface="true" scale="0.5" impactsoundtag="impact_metal_light" RequireAimToUse="True">
  <Upgrade gameversion="0.10.0.0" scale="0.5" />
  <PreferredContainer primary="medfabcab" secondary="medcontainer"/>
  <PreferredContainer secondary="abandonedmedcab,piratemedcab,wreckmedcab" minamount="1" maxamount="2" spawnprobability="0.5" />
  <Price baseprice="60">
    <Price storeidentifier="merchantoutpost" sold="false" multiplier="1.25" />
    <Price storeidentifier="merchantcity" sold="false" multiplier="1.25" />
    <Price storeidentifier="merchantresearch" multiplier="0.9" minavailable="4" />
    <Price storeidentifier="merchantmilitary" multiplier="0.9" minavailable="2" />
    <Price storeidentifier="merchantmine" sold="false" multiplier="1.1" />
    <Price storeidentifier="merchantmedical" multiplier="0.9" minavailable="4" />
  </Price>
  <InventoryIcon texture="Content/Items/InventoryIconAtlas.png" sourcerect="384,768,64,64" origin="0.5,0.5" />
  <Sprite texture="Content/Items/Medical/Medicines.png" sourcerect="379,452,27,58" depth="0.6" origin="0.5,0.5" />
  <Body width="25" height="55" density="10.2" waterdragcoefficient="1"/>
  <MeleeWeapon canBeCombined="true" removeOnCombined="true" slots="Any,RightHand,LeftHand" aimpos="40,5" handle1="0,0" holdangle="220" reload="1.0" msg="ItemMsgPickUpSelect" HitOnlyCharacters="true">
    <StatusEffect type="OnSuccess" target="This" Condition="-100.0" setvalue="true"/>
    <StatusEffect type="OnSuccess" target="UseTarget">
      <Conditional entitytype="eq Character"/>
      <Sound file="Content/Items/Medical/Syringe.ogg" range="500" />
    </StatusEffect>
    <StatusEffect type="OnImpact" target="UseTarget" multiplyafflictionsbymaxvitality="true" AllowWhenBroken="true">
      <Affliction identifier="stun" amount="0.1" />
    </StatusEffect>
    <StatusEffect tags="medical" type="OnUse" target="UseTarget" disabledeltatime="true" allowwhenbroken="true">
      <Affliction identifier="adrenalinerush" amount="10" />
    </StatusEffect>
    <!-- Remove the item when fully used -->
    <StatusEffect type="OnBroken" target="This">
      <Remove />
    </StatusEffect>
  </MeleeWeapon>
  <Projectile characterusable="false" launchimpulse="18.0" sticktocharacters="true" launchrotation="-90" inheritstatuseffectsfrom="MeleeWeapon" inheritrequiredskillsfrom="MeleeWeapon" />
</Adrenaline>
```
```
<Affliction
  identifier="adrenalinerush"
  type="adrenaline"
  isbuff="true"
  limbspecific="false"
  maxstrength="10"
  iconcolors="33,75,78;126,211,224;126,211,224;227,247,249">
  <Description
    textidentifier="afflictiondescription.adrenalinerush.self"
    target="Self" />
  <Description
    textidentifier="afflictiondescription.adrenalinerush"
    target="OtherCharacter " />
  <Effect minstrength="0" maxstrength="10" strengthchange="-1">
    <StatValue stattype="AttackMultiplier" value="0.1" />
    <AbilityFlag flagtype="AlwaysStayConscious" />
  </Effect>
  <!-- TODO Icon--><Icon texture="Content/UI/TalentsIcons4.png" sourcerect="768,256,128,128" color="10,193,114,255" origin="0,0"/>
</Affliction>
```
### Problem
We can see that it gives 10 seconds of always staying conscious. Basically, 10 seconds of invincibility. This is clearly way too powerful. 

### Original Solution
Perhaps the original idea is to use this to let the medic get some time to save the patient, but also means 10 seconds of invincibility. Imagine that in PVP a player, combined with stun resistance from things like hyperzine, is completely unkillable and unstoppable for 30 seconds. You can also hold a grenade or 40mm nuke to launch suicide attack and come back alive, and repeat this process. In single player, even in early game, you can just solo melee lots of monsters just with adrenaline. Another thing that made this matter worse is that adrenaline is considered to be a basic material, meaning it's kind of abundant and cheap. This makes them highly available for spamming. This also makes the medic's talent miracle worker, which was supposed to keep patientes alive, kind of useless. 

Now some efforts have been taken to mitigate this issue, but they aren't the best ones. For example, it has been made that adrenalies are more scarce, but this has the unintentional effect of making those medicinces that relies on adrenaline even more scarce, and without reconsidering their effects those medicines may just become too costly to use or simply useless. Also, the simple act of tunning adrenaline itself can also have balancing issues. The affliction itself is also hard to balance and inflexible. The "alwaysstayconscious" is a discrete effect: you can't stay 50% conscious. This means that you can only use the duration of the effect to, for example, make a stronger version of adrenaline.

### New Solution

Now let us consider a similar approach that makes the patient lives longer, is not overpowered, and flexible. How about we simply make this affliction give a temperorary vitality boost scaling with strength? This clearly makes the patients live longer as you have a temperorary health boost. This isn't overpowered as having some more health doesn't mean you won't die, and so we don't need to change adrenaline, which causes a chain of side effects, just to balance the overpowered effects. This is very flexible: you can use the strength to control how much vitality is provided and does it scale with max vitality; you can use duration to control how long this effect lasts.

This is can also be easily implemented using existing capabilities. Here I provided a piece of code that does exactly this:
```
<Effect MinStrength="0" MaxStrength="100" StrengthChange="-1" minvitalitydecrease="0" maxvitalitydecrease="-60"/>
```
In conclusion, instead of making adrenaline just keeps you from falling unconscious and be invincible for sometime, simply changing it to giving a temperorary health boost solves many balancing problems as well as making this drug more resonable and realstic without removing the fun part.
